apps/dashboard/src/components/layouts/hooks/use-layouts-url-state.tsx

```diff
@@ -0,0 +1,153 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { DirectionEnum } from '@novu/shared';
+
+export type LayoutsSortableColumn = 'createdAt' | 'updatedAt';
+
+export type LayoutsFilter = {
+  query: string;
+  orderBy?: LayoutsSortableColumn;
+  orderDirection?: DirectionEnum;
+  before?: string;
+  after?: string;
+};
+
+export const defaultLayoutsFilter: LayoutsFilter = {
+  query: '',
+  orderBy: 'createdAt',
+  orderDirection: DirectionEnum.DESC,
+};
+
+export type LayoutsUrlState = {
+  filterValues: LayoutsFilter;
+  handleFiltersChange: (newFilters: Partial<LayoutsFilter>) => void;
+  toggleSort: (column: LayoutsSortableColumn) => void;
+  resetFilters: () => void;
+  handleNext: () => void;
+  handlePrevious: () => void;
+  handleFirst: () => void;
+  handleNavigationAfterDelete: (afterCursor: string) => void;
+};
+
+type UseLayoutsUrlStateProps = {
+  after?: string;
+  before?: string;
+};
+
+export const useLayoutsUrlState = ({ after, before }: UseLayoutsUrlStateProps): LayoutsUrlState => {
```

Comment:
the hook that has the logic to manage the layout filters in the url search params